### ISSUE TITLE
[DeepDive] Static site generators (Jekyll, Hugo, Gatsby, Docusaurus, etc)

### ISSUE THREAD

**Issue by 0xdevalias** [0xdevalias/devalias.net#81]
(See also: https://github.com/0xdevalias/devalias.net/issues/28)

- https://www.gatsbyjs.org/
- https://gohugo.io/
- https://jekyllrb.com/
- https://docusaurus.io/
  - https://v2.docusaurus.io/
  - https://github.com/facebook/docusaurus
    - > Easy to maintain open source documentation websites.
- https://www.bridgetownrb.com/
  - https://github.com/bridgetownrb/bridgetown
    - > A Webpack-aware, Ruby-powered static site generator for the modern Jamstack era
    - > Bridgetown started life as a fork of the granddaddy of static site generators, Jekyll.
  - https://www.bridgetownrb.com/plugins/
- See also
  - https://hackernoon.com/10-static-site-generators-to-watch-in-2020-wt373yyi

## Unsorted

- https://www.staticgen.com/
  - https://github.com/netlify/staticgen
- https://headlesscms.org/
  - https://github.com/netlify/headlesscms.org

**Comment by 0xdevalias** [comment]
**tl;dr:**
- https://github.com/actions/starter-workflows/tree/main/pages
- https://github.com/actions/starter-workflows/blob/main/pages/jekyll.yml

> 👋 !
> 
> We just announced official support for building/deploying Pages straight from GitHub Actions: https://github.blog/changelog/2022-07-27-github-pages-custom-github-actions-workflows-beta/
> 
> No hacks (no `.nojekyll`, no `CNAME` files), no need to push content back to a dedicated branch, etc. It's "streamlined" and while it doesn't look like much on the surface, lots of things have been simplified underneath to make it happen.
> 
> We wrote [starter workflows for Pages for a few static site generators](https://github.com/actions/starter-workflows/blob/main/pages) including one for [plain Jekyll](https://github.com/actions/starter-workflows/blob/main/pages/jekyll.yml). By that I mean, a `Gemfile` is required and so is a `_config.yml` file but there are no "vendor locks" (in the name of security): you can depend on the latest version of Jekyll, use all plugins and all the themes you want. We don't meddle with your configuration file either because Actions is a secure sandbox.
> 
> I am aware that this is not a drop in replacement for what `pages-gem` provides today (a default theme, configuration file, etc.). This is also not exactly the objective of this new feature but we have to start somewhere! `pages-gem` today fills two goals: (a) provide the security requirements for a non-Actions build infrastructure, (b) provide default so sites can be built out of markdown files seamlessly without a user needing to even know what Jekyll is in the first place. I am not sure yet that we need `pages-gem` to answer (b) in a GitHub Actions world. I am hopefully my team or the community will figure out something soon enough.
> 
> This whole feature is in public beta today and I am very excited about it 🥳 This is certainly not perfect and will be iterated on but we want to hear your feedback. Here or on the new [community site](https://github.com/orgs/community/discussions/).
> 
> _Originally posted by @yoannchaudet in https://github.com/github/pages-gem/issues/651#issuecomment-1197448026_
